src/nvim/tui/tui.c

```diff
@@ -1807,8 +1808,10 @@ retry: \
       unibi_format(vars, vars + 26, str, params, out, tui, pad, tui); \
       if (tui->overflow) { \
         tui->bufpos = orig_pos; \
-        flush_buf(tui); \
-        goto retry; \
+        if (orig_pos > 0) { \
```

Comment:
If `orig_pos` is 0, `flush_buf(tui)` won't do anything and `goto retry;` will lead to an infinite loop. This situation can't happen when setting title as the maximum title length is much smaller than the TUI buffer size, but it can happen with a very long URL which has no size limit. #30794

Reply:
That seems worth a 1-line comment. Or more than 1 line, all of that is useful insight.

```suggestion
        /* If orig_pos is 0, `flush_buf` won't do anything and `goto retry` will loop. */ \
        if (orig_pos > 0) { \
```